addon/v1/managedclusteraddon_types.go

```diff
@@ -0,0 +1,181 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ManagedClusterAddOn is the Custom Resource object which holds the current state
+// of an addon. This object is used by addon operators to convey their state.
+// The resource should be created in the ManagedCluster's cluster namespace.
+type ManagedClusterAddOn struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// spec holds configuration that could apply to any operator.
+	// +kubebuilder:validation:Required
+	// +required
+	Spec ManagedClusterAddOnSpec `json:"spec"`
+
+	// status holds the information about the state of an operator.  It is consistent with status information across
+	// the Kubernetes ecosystem.
+	// +optional
+	Status ManagedClusterAddOnStatus `json:"status"`
+}
+
+// ManagedClusterAddOnSpec is empty for now, but you could imagine holding information like "pause".
+type ManagedClusterAddOnSpec struct {
+	// UpdateApproved represents that user has approved the update of the particular addon for a specific
+	// the managed cluster on the hub.
+	// The default value is false, it can only be set to true when the latestVersion and currentVersion
+	// is on different version.
+	// When the value is set true, the controller/operator that manages the addon will proceed to update the addon.
+	// The value will be set to false by mutating webhook when latestVersion matches currentVersion.
+	// +required
+	UpdateApproved bool `json:"updateApproved"`
+}
+
+// ManagedClusterAddOnStatus provides information about the status of the operator.
+// +k8s:deepcopy-gen=true
+type ManagedClusterAddOnStatus struct {
+	// conditions describes the state of the operator's managed and monitored components.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +optional
+	Conditions []AddOnStatusCondition `json:"conditions,omitempty"  patchStrategy:"merge" patchMergeKey:"type"`
+
+	// addonResource is a reference  to an object that contain the configuration of the addon
+	// +required
+	AddOnResource ObjectReference `json:"addonResource"`
+
+	// LatestVersion indicates the latest available Version for the addon
+	// +optional
+	LatestVersion Release `json:"latestVersion,omitempty"`
+
+	// CurrentVersion indicates the current Version of the addon
+	// During the agent update process this field will be set to same value as latestVersion after the update has been completed
+	// +optional
+	CurrentVersion Release `json:"currentVersion,omitempty"`
+
+	// UpdateAvailable indicates if there is an version update that is available for the addon
+	// it will be set to false if currentVersion.version != latestVersion.version
+	// it will be set to true if currentVersion.version == latestVersion.version
```

Comment:
Should this be `false if currentVersion.version == latestVersion.version` and `true if currentVersion.version != latestVersion.version`